frontend/pages/hosts/details/DeviceUserPage/DeviceUserPage.tsx

```diff
@@ -102,9 +102,20 @@ const DeviceUserPage = ({
       onSuccess: (returnedHost: IHostResponse) => {
         setShowRefetchSpinner(returnedHost.host.refetch_requested);
         setIsPremiumTier(returnedHost.license.tier === "premium");
-        if (returnedHost.host.refetch_requested) {
+        setHostSoftware(returnedHost.host.software);
+        setHost(returnedHost.host);
+        setOrgLogoURL(returnedHost.org_logo_url);
+        if (returnedHost?.host.refetch_requested) {
+          // If the API reports that a Fleet refetch request is pending, we want to check back for fresh
+          // host details. Here we set a one second timeout and poll the API again using
+          // fullyReloadHost. We will repeat this process with each onSuccess cycle for a total of
+          // 60 seconds or until the API reports that the Fleet refetch request has been resolved
+          // or that the host has gone offline.
           if (!refetchStartTime) {
-            if (returnedHost.host.status === "online") {
+            // If our 60 second timer wasn't already started (e.g., if a refetch was pending when
+            // the first page loads), we start it now if the host is online. If the host is offline,
+            // we skip the refetch on page load.
```

Comment:
Turns out you added really helpful comments explaining the process that I pulled from the host details page.